### ISSUE TITLE
0.28 broke TypeCompiler.Check

### ISSUE THREAD

**Issue by m-ronchi** [sinclairzx81/typebox#402]
Hi,
trying to upgrade from 0.27.4 to 0.28.1 made `TypeCompiler.Compile` produce invalid code

repro code: 
```
import { Type, type Static } from "@sinclair/typebox";
import { TypeCompiler } from "@sinclair/typebox/compiler";

const Obj = Type.Object({
  foo: Type.Object({
    bar: Type.Record(Type.String(), Type.Number()),
  }),
});
type Obj = Static<typeof Obj>;

const Check = TypeCompiler.Compile(Obj);

console.log("code:", Check.Code());
const obj: Obj = { foo: { bar: { x: 42 } } };
console.log("check:", obj, Check.Check(obj));
```

0.27 output:
```
$ ts-node src/bug.ts
code: const local_0 = new RegExp(/^(.*)$/)
return function check(value) {
  return (
    (typeof value === 'object' && value !== null && !Array.isArray(value)) &&
    (typeof value.foo === 'object' && value.foo !== null && !Array.isArray(value.foo)) &&
    (typeof value.foo.bar === 'object' && value.foo.bar !== null && !Array.isArray(value.foo.bar) && !(value.foo.bar instanceof Date) && !(value.foo.bar instanceof Uint8Array)) &&
    (Object.getOwnPropertyNames(value.foo.bar).every(key => local_0.test(key))) &&
    Object.values(value.foo.bar).every(value => ((typeof value === 'number' && Number.isFinite(value))))
 )
}
check: { foo: { bar: { x: 42 } } } true
```

0.28 output:
```
$ ts-node src/bug.ts
code: const local_0 = new RegExp(/^(.*)$/)
return function check(value) {
  return (
    (typeof value === 'object' && value !== null && !Array.isArray(value)) &&
    (typeof value.foo === 'object' && value.foo !== null && !Array.isArray(value.foo)) &&
    (typeof value.foo.bar === 'object' && value.foo.bar !== null && !Array.isArray(value.foo.bar) && !(value.foo.bar instanceof Date) && !(value.foo.bar instanceof Uint8Array)) &&
    (Object.entries(value.foo.bar).every(([key, value]) => (local_0.test(key) ? ((typeof value.foo.bar === 'number' && Number.isFinite(value.foo.bar))) : true)))
 )
}
TypeError: Cannot read properties of undefined (reading 'bar')
    at eval (eval at Compile (/Users/***/node_modules/@sinclair/typebox/compiler/compiler.js:555:45), <anonymous>:9:100)
    at Array.every (<anonymous>)
    at TypeCheck.check [as checkFunc] (eval at Compile (/Users/***/node_modules/@sinclair/typebox/compiler/compiler.js:555:45), <anonymous>:9:36)
    at TypeCheck.Check (/Users/***/node_modules/@sinclair/typebox/compiler/compiler.js:55:21)
    at Object.<anonymous> (/Users/***/src/bug.ts:15:34)
    at Module._compile (node:internal/modules/cjs/loader:1254:14)
    at Module.m._compile (/Users/***/.nvm/versions/node/v18.15.0/lib/node_modules/ts-node/src/index.ts:1618:23)
    at Module._extensions..js (node:internal/modules/cjs/loader:1308:10)
    at Object.require.extensions.<computed> [as .ts] (/Users/***/.nvm/versions/node/v18.15.0/lib/node_modules/ts-node/src/index.ts:1621:12)
    at Module.load (node:internal/modules/cjs/loader:1117:32)
```

**Comment by sinclairzx81** [comment]
@m-ronchi Hey!

Whoops, good spotting! Have fixed this up on `0.28.2`. There were some changes the Record codegen on `0.28.2` but appears the tests weren't checking for deep property pathing. 

Thanks for reporting. You're exceptional at finding breaking changes! :D It's super appreciated.
Regards
S

**Comment by sinclairzx81** [comment]
@m-ronchi Heya, going to close this one out as I think the issue should be resolved. 

Relevant Unit Test for this issue can be found here:

https://github.com/sinclairzx81/typebox/blob/master/test/runtime/compiler/record.ts#L5-L21

If you spot anything else out of order with the Record implementation, please let me know (the Record type (like Intersect) has proven to be increasingly difficult get right)

All the best!
S
